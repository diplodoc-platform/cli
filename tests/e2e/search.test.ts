import {describe, it} from 'vitest';

import {TestAdapter, compareDirectories, getTestPaths} from '../fixtures';

describe('Local search', () => {
    it('internal', async () => {
        const {inputPath, outputPath} = getTestPaths('mocks/search');

        await TestAdapter.testBuildPass(inputPath, outputPath, {
            md2md: false,
            md2html: true,
            args: '-j2 --search --interface-toc',
        });
        await compareDirectories(outputPath);
    });
});
