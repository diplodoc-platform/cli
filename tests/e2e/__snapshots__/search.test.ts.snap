// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Local search > internal > filelist 1`] = `
"[
  ".yfm",
  "_bundle/search-async-0",
  "_bundle/app-css-1",
  "_bundle/app-js-1",
  "_bundle/app-css-3",
  "_bundle/latex-extension.css",
  "_bundle/latex-extension.js",
  "_bundle/mermaid-extension.js",
  "_bundle/page-constructor-extension.css",
  "_bundle/page-constructor-extension.js",
  "_bundle/search-js-0",
  "_bundle/search-css-0",
  "_bundle/search-js-1",
  "_bundle/search-css-3",
  "_bundle/search-css-1",
  "_bundle/search-js-2",
  "_bundle/search-css-2",
  "_search/api.js",
  "_search/en/hash-resources.js",
  "_search/en/hash-index.js",
  "_search/en/hash-registry.js",
  "_search/en/index.html",
  "index.html",
  "toc.js"
]"
`;

exports[`Local search > internal 1`] = `
"allowHTML: true
langs: ['en','ru']

interface:
  toc-header: false
  favicon-src: https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico

search:
  provider: local
  tolerance: 2
  confidense: phrased
  "
`;

exports[`Local search > internal 2`] = `
"<!DOCTYPE html>
<html lang="en" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="./" />
        <title>Header</title>
        <meta name="generator" content="Diplodoc Platform vDIPLODOC-VERSION">
        <meta http-equiv="Content-Security-Policy" content="worker-src 'self';">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
        <link rel="icon" type="image/x-icon" href="https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico">
        
        <link type="text/css" rel="stylesheet" href="_bundle/search-css-1"/>
        <link type="text/css" rel="stylesheet" href="_bundle/app-css-1"/>
    </head>
    <body class="g-root g-root_theme_light">
        <div id="root"></div>
        <script type="application/json" id="diplodoc-state">
            {"data":{"leading":false,"html":"&lt;p&gt;Content&lt;/p&gt;/n","meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"}]},"headings":[],"title":"Header"},"router":{"pathname":"index","depth":1,"base":"./"},"lang":"en","langs":["en","ru"],"viewerInterface":{"toc-header":false,"favicon-src":"https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico","toc":false,"search":true,"feedback":true},"search":{"provider":"local","tolerance":2,"confidense":"phrased","enabled":true,"api":"_search/api.js","link":"_search/en/index.html"}}
        </script>
        <script type="application/javascript">
            const data = document.querySelector('script#diplodoc-state');
            window.__DATA__ = JSON.parse((function unescape(string) {
                return string.replace(/&lt;/g, "<").replace(/&gt;/g, ">").replace(/&amp;/g, "&");
              })(data.innerText));
            window.STATIC_CONTENT = false;
        </script>
        <script type="application/javascript" defer src="toc.js"></script>
        <script type="application/javascript" defer src="_search/en/hash-resources.js"></script>
        <script type="application/javascript" defer src="_bundle/search-js-0"></script>
        <script type="application/javascript" defer src="_bundle/app-js-1"></script>
        <script type="application/javascript" defer src="_bundle/search-js-2"></script>
        
    </body>
</html>"
`;

exports[`Local search > internal 3`] = `"window.__DATA__.data.toc = {"href":"index.html","path":"toc.yaml","id":"UUID"};"`;
